lib/ansible/utils/path.py

```diff
@@ -18,13 +18,16 @@
 __metaclass__ = type
 
 import os
+import re
 
 from errno import EEXIST
 from ansible.errors import AnsibleError
 from ansible.module_utils._text import to_bytes, to_native, to_text
 
 
-__all__ = ['unfrackpath', 'makedirs_safe']
+__all__ = ['unfrackpath', 'makedirs_safe', 'RE_ABS_PATH']
+
+RE_ABS_PATH = re.compile('^(?:~|%s)' % (os.path.sep))
```

Comment:
I recommend adding some unit tests to directly test any defined regexes. Just makes it easier to test if you have to change it (and also provides example hits/misses). Granted, this is not a complex regex...